src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp

```diff
@@ -270,6 +270,34 @@ NFTokenAcceptOffer::preclaim(PreclaimContext const& ctx)
         }
     }
 
+    // Fix a bug where the transfer of an NFToken with a transfer fee could
+    // give the NFToken issuer an undesired trust line.
+    if (ctx.view.rules().enabled(fixNFTokenTrustlineSurprise))
+    {
+        std::shared_ptr<SLE const> const& offer = bo ? bo : so;
+        if (!offer)
+            // Should be caught in preflight.
+            return tecINTERNAL;
+
+        // - If the NFToken has a transfer fee, and
+        // - If the NFToken doesn't have the flagCreateTrustLines flag set, and
+        // - If the Amount is not XRP, and
+        // - If the NFToken issuer is not the Amount issuer, and
+        // - If the NFToken issuer does not have a trust line for the Amount
+        // - Then reject the token accept.
+        uint256 const& tokenID = offer->at(sfNFTokenID);
+        STAmount const& amount = offer->at(sfAmount);
+        if (nft::getTransferFee(tokenID) != 0 &&
+            (nft::getFlags(tokenID) & nft::flagCreateTrustLines) == 0 &&
```

Comment:
Since the flagCreateTrustLines flag has been deprecated, would there be any consequences if we don't add this check? 

I think checking for the flag makes sense, I would just like to learn more about what kind of scenario this check would guard against. I'm thinking that some nodes may decide to enable `NonFungibleTokensV1_1` and `fixNFTokenTrustlineSurprise` amendments, but not enabling
 `fixRemoveNFTokenAutoTrustLine` for some reason? Just guessing here.

Reply:
Hi @shawnxie999, great question.  The `fixRemoveNFTokenAutoTrustLine` amendment did not deprecate `nft::flagCreateTrustLines`.  But it did deprecate the transaction flag `tfTrustLine`.  So after the amendment went live on October 27, 2022, new NFTokens could not be minted with `nft::flagCreateTrustLines` set.  That code lives here: https://github.com/XRPLF/rippled/blob/develop/src/ripple/app/tx/impl/NFTokenMint.cpp#L56-L58

But there may be any number of NFTokens minted before October 27, 2022 that _do_ have the `nft::flagCreateTrustLines` flag set.  The code needs to continue to respect the flags on those older NFTokens.  That's why we need to continue to honor `nft::flagCreateTrustLines`.

Reply:
Thanks!